File: src/liveplugin/toolwindow/settingsmenu/RunAllPluginsOnIDEStartAction.java
package liveplugin.toolwindow.settingsmenu;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.ToggleAction;
import com.intellij.openapi.project.DumbAware;
import liveplugin.Settings;

public class RunAllPluginsOnIDEStartAction extends ToggleAction implements DumbAware {
	public RunAllPluginsOnIDEStartAction() {
		super("Run All Live Plugins on IDE Start");
	}

	@Override public boolean isSelected(AnActionEvent event) {
		return Settings.getInstance().runAllPluginsOnIDEStartup;
	}

	@Override public void setSelected(AnActionEvent event, boolean state) {
		Settings.getInstance().runAllPluginsOnIDEStartup = state;
	}
}

File: src/main/liveplugin/toolwindow/settingsmenu/RunAllPluginsOnIDEStartAction.kt
package liveplugin.toolwindow.settingsmenu

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.ToggleAction
import com.intellij.openapi.project.DumbAware
import liveplugin.Settings

class RunAllPluginsOnIDEStartAction: ToggleAction("Run All Live Plugins on IDE Start"), DumbAware {
    override fun isSelected(event: AnActionEvent) =
        Settings.instance.runAllPluginsOnIDEStartup

    override fun setSelected(event: AnActionEvent, state: Boolean) {
        Settings.instance.runAllPluginsOnIDEStartup = state
    }
}
